sql/core/src/test/scala/org/apache/spark/sql/streaming/FileStreamSinkSuite.scala

```diff
@@ -478,4 +481,48 @@ class FileStreamSinkSuite extends StreamTest {
       checkDatasetUnorderly(outputDf, 1, 2, 3)
     }
   }
+
+  test("cleanup incomplete output for aborted task") {
+    withSQLConf(("spark.sql.streaming.commitProtocolClass",
+      classOf[TaskFailToCommitManifestFileCommitProtocol].getCanonicalName)) {
+      withTempDir { tempDir =>
+        val checkpointDir = new File(tempDir, "chk")
+        val outputDir = new File(tempDir, "output")
+        val inputData = MemoryStream[Int]
+        inputData.addData(1, 2, 3)
+        val q = inputData.toDF()
```

Comment:
you can call `inputData.toDS.map(_ / 0)` to fail a task rather than adding a weird ManifestFileCommitProtocol.